Tests/A005.ArmTemplates.azure-search.Tests.ps1

```diff
@@ -0,0 +1,24 @@
+# common variables
+$ResourceGroupName = "dfc-test-template-rg"
+$TemplateFile = "$PSScriptRoot\..\ArmTemplates\azure-search.json"
+
+Describe "Azure Search Deployment Tests" -Tag "Acceptance" {
+  
+  Context "When Azure Search is deployed with just search name" {
+    $TemplateParameters = @{
+      azureSearchName = "dfc-foo-bar-sch"
+    }
+    $TestTemplateParams = @{
+      ResourceGroupName       = $ResourceGroupName
+      TemplateFile            = $TemplateFile
+      TemplateParameterObject = $TemplateParameters
+    }
+
+    $output = Test-AzureRmResourceGroupDeployment @TestTemplateParams
+  
+    It "Should be deployed successfully" {
+      $output | Should -Be $null
+    }
+
+  }

```

Comment:
Should test with non-default values as well.  (I've not ran this)

  Context "When Azure Search is deployed with just search name" { 
    $TemplateParameters = @{ 
      azureSearchName = "dfc-foo-bar-sch" 
      azureSearchReplicaCount = 2
      azureSearchPartitionCount = 2
    } 
    $TestTemplateParams = @{ 
      ResourceGroupName       = $ResourceGroupName 
      TemplateFile            = $TemplateFile 
      TemplateParameterObject = $TemplateParameters 
    } 
 
    $output = Test-AzureRmResourceGroupDeployment @TestTemplateParams 
 
    It "Should be deployed successfully" { 
      $output | Should -Be $null 
    } 
 
  }

Reply:
Added a couple more tests. I've not tested every possible value of every parameter, just enough to be confident it works.